.github/workflows/run-benchmarks.yml

```diff
@@ -0,0 +1,64 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Run benchmarks
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        astropy-version: [4.0.*]
+        python-version: [3.6]  # For Python 3.7+ see https://github.com/astropy/coordinates-benchmark/issues/67
+        numpy-version: [1.17.4]
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install "numpy==$NUMPY_VERSION"
+        python -m pip install "astropy==$ASTROPY_VERSION"
+        # Non declared build dependencies
+        python -m pip install Cython
+        # Tools to benchmark
+        python -m pip install -r requirements.txt
+        # Prefetch as workaround for https://github.com/skyfielders/python-skyfield/issues/262
+        wget -c "$(python -c 'from coordinates_benchmark.tools.skyfield import EPHEMERIS; print(EPHEMERIS)')"
+      env:
+        NUMPY_VERSION: ${{ matrix.numpy-version }}
+        ASTROPY_VERSION: ${{ matrix.astropy-version }}
+    - name: Run benchmarks
+      run: |        
+        ./make.py --help
+        ./make.py tool-info
+        
+        ./make.py make_observer_table
+        ./make.py make_skycoord_table
+        ./make.py benchmark-celestial
+        ./make.py benchmark-horizontal
+        ./make.py summary-celestial
+        
+        # Making all plots is slow, but this is run rarely enough that it's fine to just do them all. However,
+        # if you are debugging, you could instead add the --tools=astropy flag to generate just some of the
+        # plots.
+        ./make.py plots
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: [benchmark]
+    if: ${{ github.ref == 'refs/heads/master' }}
+    steps:
+    - name: Deploy results
+      run: |
+        # This is a placeholder because publishing is currently broken,
+        # see https://github.com/astropy/coordinates-benchmark/issues/69
```

Comment:
Perhaps a different PR: I wonder if you can get rid of `deploy.py` altogether. I think @adrn figured out a way to deploy to gh-pages using Actions but I might also be misremembering.

Reply:
Also ping @tepickering since he is the Actions guru.

Reply:
Yes! Check out: https://github.com/adrn/astropy-sustainability/blob/main/.github/workflows/run.yml#L61

I've been using: `JamesIves/github-pages-deploy-action@3.5.7`

Reply:
also, if you do want to use `conda`, it is pre-installed on all actions runners as `$CONDA/bin/conda`. i find `pip` easier for handling python dependencies, but sometimes you need something outside that and `conda` does give you a platform-independent way of doing it (unlike `apt`).

Reply:
Thanks a lot @adrn and @tepickering for the comments! I gave up on conda already, I think it's not necessary for this project. But I also considered removing `deploy.py` and replacing it with something else if possible. I will do it in another PR though.